cluster/validation.go

```diff
@@ -316,17 +316,22 @@ func validateNetworkImages(c *Cluster) error {
 			return errors.New("flannel image is not populated")
 		}
 	} else if c.Network.Plugin == CalicoNetworkPlugin {
-		if len(c.SystemImages.CalicoCNI) == 0 {
-			return errors.New("calico cni image is not populated")
-		}
-		if len(c.SystemImages.CalicoCtl) == 0 {
-			return errors.New("calico ctl image is not populated")
-		}
-		if len(c.SystemImages.CalicoNode) == 0 {
-			return errors.New("calico image is not populated")
-		}
-		if len(c.SystemImages.CalicoControllers) == 0 {
-			return errors.New("calico controllers image is not populated")
+		// If TigeraOperator is specified then the versions used will be
+		// bundled with the Operator, so only check the Calico images when
+		// the Operator is not used.
+		if len(c.SystemImages.TigeraOperator) == 0 {
```

Comment:
There may be times when you would want to deploy Calico in the "legacy" manner rather than using the operator, but if the corresponding K8s version has the operator image defined then this sanity check may miss image misconfigurations.

Reply:
Sorry if I've misunderstood how things work with RKE but is it possible to install with an 'older' set of manifests but using a 'newer' set of images?
I thought the images and templates were both driven off the k8s version, is that not correct?
Something I should clarify is that when using the TigeraOperator there is no need for the other Calico component images to be specified as they are bundled in the operator. 
Do you have a suggestion on what this check should look like? Basically for Calico installs using the operator, only the operator is needed (none of the Calico images) and older installs only need the Calico images.

Reply:
I'm not sure that forcing all users of the Calico network plugin to use the operator past a certain K8s version makes sense, given that we have a perfectly suitable mechanism to install Calico as it is. Generally, when a hard change is introduced like this into RKE, we provide the option to use a method that is already around. `kube-dns` to `CoreDNS` is a great example of this.

Reply:
I've made updates to allow sticking with the "Legacy" Calico manifest install method if desired.